src/common/pretty_binary.cc

```diff
@@ -0,0 +1,95 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "pretty_binary.h"
+#include <stdexcept>
+#include <sstream>
+
+std::string pretty_binary_string_reverse(const std::string& pretty)
+{
+  size_t i = 0;
```

Comment:
nit, could use a `string::const_iterator` instead.

Reply:
```i``` is used to signal parsing error, and also there is pretty[i + 1] reference.
Both can be modified, but you think it will look better?

Reply:
no sure, `i` is probably more straightforward.